mithril-common/src/cardano_block_scanner/scanned_block.rs

```diff
@@ -19,11 +18,11 @@ pub struct ScannedBlock {
 
 impl ScannedBlock {
     /// Scanned block factory
-    pub fn new<T: Into<TransactionHash>, U: Into<BlockHash>>(
-        block_hash: U,
+    pub fn new<BlkHash: Into<Vec<u8>>, TxHash: Into<TransactionHash>>(
```

Comment:
It's not very clear that `BlkHash`and `TxHash` are generic types:
```suggestion
    pub fn new<BLOCK_HASH: Into<Vec<u8>>, TX_HASH: Into<TransactionHash>>(
```